### PR TITLE
Use effects and coeffects of Cextcall in selectgen

### PR DESCRIPTION
**Pull request by gretay-js**
on top of #12, only the new last commit is new. 

Currently restricted to builtin=true.

### CODE REVIEW

File: backend/selectgen.ml
Comment (mshinwell) [inline backend/selectgen.ml:344]: I'm nervous about making this change until the `c_builtin` field is here.  Can we please revert these four lines?

**Review by mshinwell — CHANGES_REQUESTED** [review]

File: backend/selectgen.ml
Comment (gretay-js) [reply backend/selectgen.ml:344]: ah, the comment about "inlined ops" is no longer relevant, but this new case in `is_simple_expr` is no different from the new case below in `effects_of` method - are you nervous about both of them? Also, the first version this PR had a constraint on builtin=true here, but I thought that you said it is not needed. Should I put it back?
There is no harm in merging this change as is, because the effects and coeffects are set conservatively everywhere at the moment. Equally, we can hold off on this PR until we merge the one that more precise effects and coeffects. The existing situation is safe.

Comment (mshinwell) [reply backend/selectgen.ml:416]: If the comment's no longer relevant, can it be removed?

Comment (gretay-js) [reply backend/selectgen.ml:416]: I've rephrase the comment to talk about Cextcall (instead of the non-exiting inlined_ops), I think it's accurate and useful.

File: backend/selectgen.ml
Comment (xclerc) [inline backend/selectgen.ml:354]: (The naming feels confusing: `effect_only` and `coeffect_only`
are projections, but `effect_and_coeffect` is not.)

File: backend/selectgen.ml
Comment (xclerc) [inline backend/selectgen.ml:414]: ("with neither (...) nor (...)"?)

**Review by xclerc — APPROVED** [review]

File: backend/selectgen.ml
Comment (gretay-js) [reply backend/selectgen.ml]: fixed.

File: backend/selectgen.ml
Comment (gretay-js) [reply backend/selectgen.ml:355]: All three are constructors for `t`.

Comment (xclerc) [reply backend/selectgen.ml:355]: Indeed... I was confused.

File: backend/selectgen.ml
Comment (xclerc) [inline backend/selectgen.ml:367]: Isn't here that we have two projections and a builder
with the same naming?

Comment (gretay-js) [reply backend/selectgen.ml]: wow, this is incredibly confusing, thanks for pointing it out! I've renamed the offender.

**Review by mshinwell — APPROVED** [review]

**Pull request merged**
